### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by maxinelasp**
# Change Summary
Adding sphinx and readthedocs to match imap_processing.

## Overview
Mostly this was created with Sphinx autogeneration. Changed by me files are listed below. Most of them were copied from imap_processing. 

## New Dependencies

sphinx, myst-parser, pydata-sphinx-theme

## New Files

- docs/README.md
   - Overview of info
- docs/source/conf.py
   - mostly copied from imap_processing
- .readthedocs.yaml

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,4 +37,9 @@ awscliv2.zip
...
 
 # Individual development CDK apps
-app*dev.py
\ No newline at end of file
+app*dev.py
+
+# Sphinx documentation
+docs/_build/
+
+requirements.txt
```
Comment (greglucas) [inline .gitignore]: I'm not sure this file needs to be updated.

The docs build is going into `docs/build` without the underscore, so I don't think that one is needed. I also don't think we want to ignore `requirements.txt` generally, but we also aren't using it on this project. So both of these are inconsequential but seem unrelated to the PR.

File: .readthedocs.yaml
Comment (greglucas) [inline .readthedocs.yaml:1]: Did you create the location on readthedocs or does that still need to be done?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Using poetry:
+
+```bash
+# Install the optional documentation dependencies
+poetry install -E docs
+# Make the html documentation
+cd docs/source
+poetry run sphinx-build . build
+# View the documentation
+open build/html/index.html
```
Comment (greglucas) [inline docs/README.md]:
```suggestion
cd docs/source
poetry run sphinx-build . build
# View the documentation
open ../build/html/index.html
```
I think you'll need to go up one level to get back to `build`

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Using pip:
+
+```bash
+# Install the optional documentation dependencies
+pip install .[doc]
+# Make the html documentation
+make -C docs html
+# View the documentation
+open docs/_build/html/index.html
```
Comment (greglucas) [inline docs/README.md]:
```suggestion
open docs/build/html/index.html
```

File: docs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+   :maxdepth: 2
+   :caption: Contents:
+
+
+
+Indices and tables
+==================
+
+* :ref:`genindex`
+* :ref:`modindex`
```
Comment (greglucas) [inline docs/source/index.rst]:
```suggestion
* :ref:`genindex`
```

Would we expect some content to be filled in these indexes by default or do we have to do that manually? The `genindex` is empty, but the `modindex` doesn't even exist locally for me so it is a 404 when I click on it.

**Review by greglucas — APPROVED** [review]
Looks good and builds locally for me! Just a few minor comments. Feel free to address the content comments in follow-up PRs.

File: .gitignore (hunk 1 above)
Comment (maxinelasp) [reply .gitignore]: The requirements.txt change is actually a holdover from a previous version that is not necessary anymore, so I will remove it. The `_build` is also from the previous version, so I will update it.

File: .readthedocs.yaml
Comment (maxinelasp) [reply .readthedocs.yaml:1]: That still needs to be done!

File: docs/source/index.rst (hunk 4 above)
Comment (maxinelasp) [reply docs/source/index.rst]: Removed that whole section

**Review by bourque — APPROVED** [review]
